lib/qops/environment.rb

```diff
@@ -31,20 +31,75 @@ def self.print_with_colour(message, level = :normal)
       end
     end
 
-    def initialize
-      %w[deploy_type region stack_id app_name].each do |v|
+    def initialize(profile: nil)
+      @_aws_config = { region: configuration.region }
+      @_aws_config[:profile] = profile unless profile.nil?
+      puts Rainbow("using aws profile #{profile}").bg(:black).green unless profile.nil?
+
+      %w[deploy_type region app_name].each do |v|
         fail "Please configure #{v} before continuing." unless option?(v)
       end
 
+      fail 'Please configure stack_id or stack_name before continuing' unless option?('stack_id') || option?('stack_name')
+
       begin
-        opsworks.config.credentials.credentials
+        opsworks.config.credentials.credentials unless profile
       rescue => e # rubocop:disable Lint/RescueWithoutErrorClass
         raise "There may be a problem with your aws credentials. Please correct with `aws configure`. Error: #{e}"
       end
     end
 
-    def application_id
-      configuration.application_id unless configuration.application_id.blank?
+    def identity_from_config
+      configuration.stack_id ? :stack_id : :stack_name
+    end
+
+    def find_stack(options = {})
+      return @_stack if @_stack
+      key = if !options[:name].nil?
+              :name
+            elsif !options[:stack_id].nil?
+              :stack_id
+            else
+              identity_from_config
```

Comment:
could you put print statement to notify user they are using the opswork config here?